internal/aws/containerinsight/utils.go

```diff
@@ -0,0 +1,230 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package containerinsight
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+// aggregate fileds
+func AggregateFields(fields []map[string]interface{}) map[string]float64 {
+	if len(fields) == 0 {
+		return nil
+	}
+
+	result := make(map[string]float64)
+	// Use the first element as the base
+	for k, v := range fields[0] {
+		result[k] = v.(float64)
+	}
+
+	if len(fields) == 1 {
+		return result
+	}
+
+	for i := 1; i < len(fields); i++ {
+		for k, v := range result {
+			if fields[i][k] == nil {
+				continue
+			}
+			result[k] = v + fields[i][k].(float64)
+		}
+	}
+	return result
+}
+
+func IsNode(mType string) bool {
+	return mType == TypeNode || mType == TypeNodeNet || mType == TypeNodeFS || mType == TypeNodeDiskIO
+}
+func IsInstance(mType string) bool {
+	return mType == TypeInstance || mType == TypeInstanceNet || mType == TypeInstanceFS || mType == TypeInstanceDiskIO
+}
+func IsContainer(mType string) bool {
+	return mType == TypeContainer || mType == TypeContainerDiskIO || mType == TypeContainerFS
+}
+func IsPod(mType string) bool {
+	return mType == TypePod || mType == TypePodNet
+}
+
+func getPrefixByMetricType(mType string) string {
+	prefix := ""
+	instancePrefix := "instance_"
+	nodePrefix := "node_"
+	instanceNetPrefix := "instance_interface_"
+	nodeNetPrefix := "node_interface_"
+	podPrefix := "pod_"
+	podNetPrefix := "pod_interface_"
+	containerPrefix := "container_"
+	service := "service_"
+	cluster := "cluster_"
+	namespace := "namespace_"
+
+	switch mType {
+	case TypeInstance:
+		prefix = instancePrefix
+	case TypeInstanceFS:
+		prefix = instancePrefix
+	case TypeInstanceDiskIO:
+		prefix = instancePrefix
+	case TypeInstanceNet:
+		prefix = instanceNetPrefix
+	case TypeNode:
+		prefix = nodePrefix
+	case TypeNodeFS:
+		prefix = nodePrefix
+	case TypeNodeDiskIO:
+		prefix = nodePrefix
+	case TypeNodeNet:
+		prefix = nodeNetPrefix
+	case TypePod:
+		prefix = podPrefix
+	case TypePodNet:
+		prefix = podNetPrefix
+	case TypeContainer:
+		prefix = containerPrefix
+	case TypeContainerDiskIO:
+		prefix = containerPrefix
+	case TypeContainerFS:
+		prefix = containerPrefix
+	case TypeService:
+		prefix = service
+	case TypeCluster:
+		prefix = cluster
+	case TypeClusterService:
+		prefix = service
+	case TypeClusterNamespace:
+		prefix = namespace
+	case K8sNamespace:
+		prefix = namespace
+	default:
+		log.Printf("E! Unexpected MetricType: %s", mType)
+	}
+	return prefix
+}
+
+func MetricName(mType string, name string) string {
```

Comment:
Please add godoc for these exported functions.

Reply:
Sure.